File: MissionRocketLaunch/MissionTeegardenB/app/src/main/java/com/flame4ost/missionteegardenb/U1.java
package com.flame4ost.missionteegardenb;

public class U1 extends Rocket{

    public U1 (){
        cost = 100;
        weight = 10000;
        max_weight = 18000;
        launch_explosion = 0.0;
        landing_crash = 0.0;
        current_weight = weight;
    }

    @Override
    public boolean land(){
        int random = (int)(Math.random() * 100 + 1);
        this.landing_crash = 1.0 * (this.current_weight - this.weight) / (this.max_weight - this.weight);

        return this.landing_crash <= random;
    }

    @Override
    public boolean launch(){
        int random = (int)(Math.random() * 100 + 1);
        this.launch_explosion = 5.0 * (this.current_weight - this.weight) / (this.max_weight - this.weight);

        return this.launch_explosion <= random;
    }
}

File: MissionRocketLaunch/MissionTeegardenBKotlin/app/src/main/java/com/flame4ost/missionteegardenbkotlin/U1.kt
package com.flame4ost.missionteegardenbkotlin

class U1: Rocket() {
    init {
        cost = 100
        weight = 10000
        max_weight =18000
        launch_explosion = 0.0
        landing_crash = 0.0
        current_weight = weight
    }

    override fun land(): Boolean{
        val random = (Math.random() * 100 + 1).toInt()
        landing_crash = 1.0 * (current_weight - weight) / (max_weight - weight)

        return landing_crash <= random
    }

    override fun launch(): Boolean {
        val random = (Math.random() * 100 + 1).toInt()
        launch_explosion = 5.0 * (current_weight - weight) / (max_weight - weight)

        return launch_explosion <= random
    }


}